paasta_itests/itest_utils.py

```diff
@@ -29,19 +30,26 @@
 
 
 def update_context_marathon_config(context):
-    whitelist_keys = {'id', 'backoff_factor', 'backoff_seconds', 'max_instances', 'mem', 'cpus', 'instances'}
+    whitelist_keys = {
+        'id', 'backoff_factor', 'backoff_seconds', 'max_instances', 'mem', 'cpus', 'instances',
+        'marathon_shard', 'previous_marathon_shards',
+    }
     with mock.patch.object(
         MarathonServiceConfig, 'get_min_instances', autospec=True, return_value=1,
     ), mock.patch.object(
         MarathonServiceConfig, 'get_max_instances', autospec=True,
     ) as mock_get_max_instances:
         mock_get_max_instances.return_value = context.max_instances if 'max_instances' in context else None
+        service_configuration_lib._yaml_cache = {}
```

Comment:
Why do we even use `service_configuration_lib.read_extra_service_information`? I feel like the magic yaml caching only ever bites us.